include/fmt/format-inl.h

```diff
@@ -768,6 +768,12 @@ void sprintf_format(Double value, internal::buffer<char>& buf,
   *format_ptr++ = type;
   *format_ptr = '\0';
 
+#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
+  if (spec.precision > 1000) {
```

Comment:
I kept this, because I think it is difficult to prevent the fuzzer from causing this input. That means one would have to write a format parser, to know what to reject. And that would be a duplicate of fmt internals. So this preprocessor check is neccessary. Unless it is possible to indirectly stop it by rejecting the buffer to be resized to large values?
The only downside to keeping this, is that it is ugly and parsing is a (tiny) bit slower.

Reply:
You can prevent large allocations with `format_to` by creating a class that derives from `memory_buffer` and overriding `grow`:

```c++
class limited_memory_buffer : public fmt::memory_buffer {
 protected:
  void grow(size_t capacity) {
    if (capacity > 1000) throw std::runtime_error("large allocation");
    fmt::memory_buffer::grow(capacity);
  }
};

limited_memory_buffer buf;
format_to(buf, "{:.2000}", 4.2); // throws
```

This will only work with `format_to`, but `format` uses `(v)format_to` internally and then copies the result into `std::string` so there is probably no need to fuzz both.

Reply:
That is a nice solution!
I implemented it as proposed, but it won't help because for instance 
```cpp
basic_writer<Range>::write_double
```
uses a memory_buffer on the stack. ifdefing that to be fuzzer aware, is in my opinon worse than having the ifdef inside grow() to begin with.

Do you still want me to carry on in that direction?

Reply:
> uses a memory_buffer on the stack

Ah, indeed. Let's go with your current solution but try to reduce the number of `#ifdef`s. Is it enough to check the size only at the beginning of `basic_memory_buffer::grow`?

Reply:
I could not remove any of the checks without having the fuzzer consume too much resources.